numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -2851,13 +2851,6 @@ npyiter_allocate_arrays(NpyIter *iter,
             npyiter_replace_axisdata(iter, iop, op[iop], ondim,
                     PyArray_DATA(op[iop]), op_axes ? op_axes[iop] : NULL);
 
-            /*
-             * New arrays are guaranteed true-aligned, but copy/cast code
-             * needs uint-alignment in addition.
-             */
-            if (IsUintAligned(out)) {
-                op_itflags[iop] |= NPY_OP_ITFLAG_ALIGNED;
```

Comment:
I'm confused why we don't simply revert to the code pre-#12626, and set `OP_ITFLAG_ALIGNED` unconditionally (and then use it in the transfer function `if` statements - it seems the least amount of churn and keeps the flag.